app/api/unit_activity_sets_api.rb

```diff
@@ -0,0 +1,82 @@
+require 'grape'
+
+module Api
+  class UnitActivitySetsApi < Grape::API
+    helpers AuthenticationHelpers
+    helpers AuthorisationHelpers
+
+    before do
+      authenticated?
+    end
+
+    desc 'Add an unit activity set'
+    params do
+      requires :activity_type_id, type: Integer, desc: 'The id of the activity type'
+    end
+    post '/units/:unit_id/activity_sets' do
+      unit = Unit.find(params[:unit_id])
+      unless authorise? current_user, unit, :update
+        error!({ error: 'Not authorised to update this unit' }, 403)
+      end
+
+      activity_type = ActivityType.find(params[:activity_type_id])
+      unit.add_activity_set(activity_type)
+    end
+
+    desc 'Update an unit activity set'
+    params do
+      optional :activity_type_id, type: Integer, desc: 'The id of the activity type'
+    end
+    put '/units/:unit_id/activity_sets/:id' do
+      unit = Unit.find(params[:unit_id])
+      unless authorise? current_user, unit, :update
+        error!({ error: 'Not authorised to update this unit' }, 403)
+      end
+
+      activity_type_id = params[:activity_type_id]
+      if activity_type_id.present?
+        activity_type = ActivityType.find(activity_type_id)
+      end
+
+      unit.update_activity_set(params[:id], activity_type)
+    end
+
+    desc 'Delete an unit activity set inside a unit'
+    delete '/units/:unit_id/activity_sets/:id' do
+      unit = Unit.find(params[:unit_id])
+      unless authorise? current_user, unit, :update
+        error!({ error: 'Not authorised to update this unit' }, 403)
+      end
+
+      unit.unit_activity_sets.find(params[:id]).destroy
+    end
+
+    desc "Get an unit activity set details"
+    get '/unit_activity_sets/:id' do
```

Comment:
Will we get these individually? Shouldn't these be returned within the unit itself?

Reply:
Done 👍 
https://github.com/AkshatBajaj/doubtfire-api/blob/bf715c5a7c4ec98ebacc120eec63cb3f99abed33/app/serializers/unit_serializer.rb#L50

Reply:
Do we need more information about the Activity Type in the unit serializer?

Reply:
What is serialised in the activity set? Is there a serialiser?